### PR TITLE
[Refactor] Refactored backend

### PR DESCRIPTION
**Pull request by nat212**

### CODE REVIEW

File: backend/app/__init__.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import os
+from flask import Flask
+from flask_cors import CORS
+from .views.auth import auth
```
Comment (1tsS0meth1ng) [inline backend/app/__init__.py]: Maybe make this backend.app.views.auth better convention imo

File: backend/app/views/auth.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    """
+    A method to authenticate with google.
+
+    Returns
+    -------
+    Response
+        an http response
+    """
+
+    check_endpoint = "https://www.googleapis.com/oauth2/v1/tokeninfo?access_token="
```
Comment (brendan-bath) [inline backend/app/views/auth.py:103]: We should potentially move this to a file so it can be changed out

File: backend/app/__init__.py (hunk 1 above)
Comment (brendan-bath) [inline backend/app/__init__.py]: Maybe make this backend.app.views.auth better convention imo. It's what you've done with most of the other imports

File: backend/app/views/ec2.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        return Response('Success', status=200, mimetype='application/text')
+    except ClientError as ex:
+        message, status = aws.boto3_errors(ex)
+        return Response(message, status=status, mimetype='application/text')
+
+
+@ec2.route('/api/instances/<instance_id>/restart')
+@require_auth
+@aws.boto3_client()
+def restart_instance(user, client, instance_id):
```
Comment (brendan-bath) [inline backend/app/views/ec2.py:61]: We should potentially perform a check to make sure all the parameters are supplied. However this is probably not needed.

**Review by brendan-bath — CHANGES_REQUESTED** [review]

File: backend/app/views/ec2.py (hunk 3 above)
Comment (nat212) [reply backend/app/views/ec2.py:61]: It will actually throw an error if the parameters aren't there.

File: backend/app/views/auth.py (hunk 2 above)
Comment (nat212) [reply backend/app/views/auth.py:103]: I don't think that's necessary at this point, it's unlikely to change any time soon.

**Review by brendan-bath — APPROVED** [review]

**Pull request merged**